packages/tailwind-palette/src/colors.ts

```diff
@@ -0,0 +1,120 @@
+// @ts-ignore
+import colorNamer from 'color-namer';
+
+export type Palette = {
+  name: string;
+  colors: {
+    [key: number]: string;
+  };
+};
+
+type Rgb = {
+  r: number;
+  g: number;
+  b: number;
+};
+
+function hexToRgb(hex: string): Rgb | null {
+  const sanitizedHex = hex.replace(/##/g, '#');
+  const colorParts = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(sanitizedHex);
+
+  if (!colorParts) {
+    return null;
+  }
+
+  const [, r, g, b] = colorParts;
+
+  return {
+    r: parseInt(r, 16),
+    g: parseInt(g, 16),
+    b: parseInt(b, 16),
+  } as Rgb;
+}
+
+function rgbToHex(r: number, g: number, b: number): string {
```

Comment:
https://stackoverflow.com/questions/39494689/is-it-possible-to-restrict-number-to-a-certain-range
typescript can help to restrict number range now

Reply:
@TianyiLi I think it not ready yet through the discussion?